src/naturalearth.jl

```diff
@@ -3,34 +3,264 @@
 # -----------------------------------------------------------------
 
 """
-Provides a single function `NaturalEarth.get` to download data from the
-NaturalEarth database.  Please check its docstring for more details.
+Provides utility functions to download data from the NaturalEarth database.
+Please check the docstrings for more details.
 """
 module NaturalEarth
 
 using GeoIO
+using DataDeps
+using TableTransforms
+using Tables
+using CSV
 
-import NaturalEarth as NE
+function download(scale, entity, variant)
+  if scale ∉ ("1:10", "1:50", "1:100")
+    throw(ArgumentError("invalid scale, please use one these: 1:10, 1:50, 1:100"))
+  end
 
-"""
-    NaturalEarth.get(name::String; version = v"5.1.2")
-    NaturalEarth.get(name::String, scale::Int; version = v"5.1.2")
+  table = CSV.File(joinpath(@__DIR__, "..", "artifacts", "NaturalEarth.csv"))
+
+  srows = table |> Filter(row -> row.SCALE == "$(scale)m" && contains(row.ENTITY, entity) && contains(row.VARIANT, variant))
+  srow = Tables.rows(srows) |> first
 
-Load a NaturalEarth dataset as a `GeoTable`.
+  url = srow.URL
+  fname = split(url, "/") |> last |> splitext |> first
+  ID = "NaturalEarth_$fname"
 
-The `name` should not include the `ne_` prefix, and if providing a `scale`
-should also not include a scale. No suffix should be added.
+  try
+    # if data is already on disk
+    # we just return the path
+    @datadep_str ID
+  catch
+    # otherwise we register the data
+    # and download using DataDeps.jl
+    try
+      register(DataDep(ID,
+        """
+        Geographic data provided by the https://www.naturalearthdata.com project.
+        Scale: $(srow.SCALE)
+        Entity: $(srow.ENTITY)
+        Variant: $(srow.VARIANT)
+        """,
+        url,
+        Any,
+        post_fetch_method=DataDeps.unpack
+      ))
+      @datadep_str ID
+    catch
+      throw(ErrorException("download failed due to internet and/or server issues"))
+    end
+  end
+end
 
-# Examples
+function get(scale, entity, variant; kwargs...)
+  path = download(scale, entity, variant)
 
-```julia
-NaturalEarth.get("admin_0_countries", 110)
-NaturalEarth.get("110m_admin_0_countries")
-```
-"""
-function get(args...; kwargs...)
-  table = NE.naturalearth(args...; kwargs...)
-  GeoIO.asgeotable(table)
+  # find Shapefile/GeoTIFF file
+  files = readdir(path; join=true)
+  isgeo(f) = last(splitext(f)) ∈ (".shp", ".tif", ".tiff")
+  file = files[findfirst(isgeo, files)]
+
+  GeoIO.load(file; kwargs...)
+end
+
+# -----------
+# PUBLIC API
+# -----------
+
+function countries(variant="default"; scale="1:10", kwargs...)
```

Comment:
Need docstring for each function in public api. Each docstring can list the variants.